pymeasure/instruments/toptica/ibeamsmart.py

```diff
@@ -0,0 +1,118 @@
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.toptica.adapters import TopticaAdapter
+from pymeasure.instruments.validators import strict_discrete_set, strict_range
+
+
+class IBeamSmart(Instrument):
+    """ IBeam Smart laser diode
+
+    :param port: pyvisa resource name of the instrument
+    :param baud_rate: communication speed, defaults to 115200
+    :param kwargs: Any valid key-word argument for VISAAdapter
+    """
+    version = Instrument.measurement(
+           "ver", """ Firmware version number """,
+    )
+
+    serial = Instrument.measurement(
+           "serial", """ Serial number of the laser system """,
+    )
+
+    temp = Instrument.measurement(
+            "sh temp",
+            """ temperature of the laser diode in degree centigrade.""",
+    )
+
+    system_temp = Instrument.measurement(
+            "sh temp sys",
+            """ base plate (heatsink) temperature in degree centigrade.""",
+    )
+
+    laser_enabled = Instrument.control(
+            "sta la", "la %s",
+            """ Status of the laser diode driver.
+                This can be True if the laser is on or False otherwise""",
+            validator=strict_discrete_set,
+            values=[True, False],
+            get_process=lambda s: True if s == 'ON' else False,
+            set_process=lambda v: "on" if v else "off",
+    )
+
+    channel1_enabled = Instrument.control(
+            "sta ch 1", "%s",
+            """ Status of Channel 1 of the laser.
+                This can be True if the laser is on or False otherwise""",
+            validator=strict_discrete_set,
+            values=[True, False],
+            get_process=lambda s: True if s == 'ON' else False,
+            set_process=lambda v: "en 1" if v else "di 1",
+    )
+
+    channel2_enabled = Instrument.control(
+            "sta ch 2", "%s",
+            """ Status of Channel 2 of the laser.
+                This can be True if the laser is on or False otherwise""",
+            validator=strict_discrete_set,
+            values=[True, False],
+            get_process=lambda s: True if s == 'ON' else False,
+            set_process=lambda v: "en 2" if v else "di 2",
+    )
+
+    power = Instrument.control(
+            "sh pow", "ch pow %f mic",
+            """ Actual output power in uW of the laser system. In pulse mode
+            this means that the set value might not correspond to the readback
+            one.""",
+            validator=strict_range,
+            values=[0, 200000],
+    )
+
+    def __init__(self, port, baud_rate=115200, **kwargs):
```

Comment:
I think it would be good to check that port is a string, and otherwise raise, because all this does not work with anything else than a Toptica/VISAAdapter, and people could pass a SerialAdapter in.

Reply:
done